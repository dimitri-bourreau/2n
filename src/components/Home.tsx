import React, { FunctionComponent, ReactElement, useState } from 'react';
import { RouteComponentProps } from '@reach/router';
import Introduction from './game/Introduction';
import Game from './game/Game';

const Home: FunctionComponent<RouteComponentProps> = (): ReactElement => {
  const [gameIsOn, setGameIsOn] = useState(false);

  const handlePlayGame = (game: boolean): null => {
    setGameIsOn(game);
    return null;
  };

  return (
    <div id="home">
      {gameIsOn ? (
        <Game handlePlayGame={handlePlayGame} />
      ) : (
        <Introduction handlePlayGame={handlePlayGame} />
      )}
    </div>
  );
};

export default Home;
