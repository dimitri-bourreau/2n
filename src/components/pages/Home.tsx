import React, { FunctionComponent, ReactElement } from 'react';
import { RouteComponentProps } from '@reach/router';
import Introduction from '../2digits/Introduction';
import Rules from '../2digits/Rules';

const Home: FunctionComponent<RouteComponentProps> = (): ReactElement => {
  return (
    <>
      <Introduction />
      <Rules />
    </>
  );
};

export default Home;
