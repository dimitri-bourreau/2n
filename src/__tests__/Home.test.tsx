import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Home from '../components/Home';

Enzyme.configure({ adapter: new Adapter() });

describe('<Home />', () => {
  let wrapper;
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation(init => [init, setState]);

  beforeEach(() => {
    wrapper = Enzyme.shallow(<Home />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('Launch game', () => {
    it('should set gameIsOn to true', () => {
      wrapper.find('#play-game-button').props().onClick();
      expect(setState).toHaveBeenCalledWith(true);
    });
  });
});
