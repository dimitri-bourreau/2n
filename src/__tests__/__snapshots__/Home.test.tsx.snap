// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Home component should match snapshot 1`] = `
<div
  id="home"
>
  <h1
    className="mt-5 mx-auto text-center"
  >
    2n
  </h1>
  <h6
    className="text-center"
  >
    Will you remember 2 numbers ago?
     
    <span
      aria-label="emoji-hand-mouth"
      role="img"
    >
      ðŸ¤­
    </span>
  </h6>
  <button
    className="mt-5 btn btn-success btn-lg btn-pill btn-block"
    onClick={[Function]}
    type="button"
  >
    Play
  </button>
  <div
    className="mx-auto my-5 card"
  >
    <div
      className="card-header"
    >
      The game
    </div>
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Will you remember 2 digits ago?
      </h5>
      <p>
        Every turn the game will show you one digit.
      </p>
      <p>
        First two turns just remember the digits.
      </p>
      <p>
        Then, the game will ask you if a digit was displayed two turns ago!
      </p>
      <h5>
        Example
      </h5>
      <ul>
        <li>
          Turn 1 : you remember 2.
        </li>
        <li>
          Turn 2 : you remember 7.
        </li>
        <li>
          Turn 3 : you remember 8. The game asks you if 5 was displayed two turns ago: no.
        </li>
        <li>
          Turn 4 : you remember 1. The game asks you if 7 was displayed two turns ago: yes.
        </li>
      </ul>
      <p>
        Go as far as you can, make the biggest score!
      </p>
    </div>
  </div>
</div>
`;
